pkg/lease/fake.go

```diff
@@ -0,0 +1,34 @@
+package lease
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type FakeClient struct {
+	Calls    []string
+	Failures sets.String
+}
+
+func (c *FakeClient) Acquire(rtype string) (string, error) {
+	call := "acquire " + rtype
+	c.Calls = append(c.Calls, call)
+	if c.Failures.Has(call) {
+		return "", fmt.Errorf("injected failure %q", call)
+	}
+	return rtype + strconv.Itoa(len(c.Calls)-1), nil
+}
+
+func (c *FakeClient) Release(name string) error {
+	call := "release " + name
```

Comment:
appending the name with a space between them doesn't seem right? Is this intentional?

Reply:
See `TestError` and `TestAcquireRelease` in `pkg/steps/lease_test.go`.